cpp/src/io/orc/reader_impl.hpp

```diff
@@ -48,6 +48,17 @@ struct stripe_source_mapping;
 }  // namespace
 class aggregate_orc_metadata;
 
+/**
+ * @brief Keeps track of orc mapping and child column details.
+ */
+struct reader_column_meta {
+  std::vector<std::vector<int32_t>>
+    orc_col_map;                         // Mapping between column id in orc to processing order.
+  std::vector<int32_t> num_child_rows;   // number of child rows of a list column
+  std::vector<int32_t> child_start_row;  // start row of a child column in a stripe
```

Comment:
If this contains a 2d structure then perhaps either a vector of vectors is better suited here. Or if that is not possible, some mention of this in the docs.